src/library.js

```diff
@@ -1714,25 +1714,25 @@ LibraryManager.library = {
   //    being compiled. Not sure how to tell LLVM to not do so.
   // ==========================================================================
 
+#if MAIN_MODULE == 0
+  dlopen: function(/* ... */) {
+    abort("To use dlopen, you need to use Emscripten's linking support, see https://github.com/kripken/emscripten/wiki/Linking");
+  },
+  dlclose: 'dlopen',
+  dlsym:   'dlopen',
+  dlerror: 'dlopen',
+  dladdr:  'dlopen',
+#else
+
   $DLFCN: {
     error: null,
     errorMsg: null,
-
-    // next free handle to use for a loaded dso.
-    // (handle=0 is avoided as it means "error" in dlopen)
-    nextHandle: 1,
-
-    loadedLibs: {}, // handle -> [refcount, name, lib_object]
-    loadedLibNames: {}, // name -> handle
   },
   // void* dlopen(const char* filename, int flag);
-  dlopen__deps: ['$DLFCN', '$FS', '$ENV'],
+  dlopen__deps: ['$LDSO', '$DLFCN', '$FS', '$ENV'],
```

Comment:
Note that `$LDSO` means: depend on the object `LDSO` in the library (the `$` prefix is due to a limitation in the library system). There is no need to add a dependency on something not in the libraries, so as the PR is written now these deps additions can be removed. (However, see above question on where this should be.)

(If we end up putting LDSO in the library, then the prefix `$` is needed as I mentioned for technical reasons. It's only needed in the deps and in the declaration, but when JS code calls it, it should not appear.)

Reply:
Thanks for clarifying this. Indeed, since LDSO is staying always present in the runrime (in support.js - see the other reply) we don't need to put it into `__deps` here and we don't need to use `$`. I'm amending the patch accordingly.